src/foundations/rhi/Shader.zig

```diff
@@ -0,0 +1,93 @@
+fragment_shader: fragment_shader_type,
+instance_data: bool,
+vertex_partials: [max_vertex_partials][]const u8 = undefined,
+num_vertex_partials: usize = 0,
+program: u32 = 0,
+
+const max_vertex_partials: usize = 10;
+
+pub const fragment_shader_type = enum(usize) {
```

Comment:
**suggestion:** Use more descriptive names for fragment shader types in the enum

Instead of 'color', 'normals', and 'texture', consider using more specific names like 'BasicColor', 'NormalMapping', and 'TextureMapping' to make the enum more self-documenting.